### ISSUE TITLE
Deadlock in environment cleanup

### ISSUE THREAD

**Issue by wouterdb** [inmanta/inmanta-core#6117]
```
09:52:08.330 INFO 200 GET /api/v1/agent (127.0.0.1) 7.16ms
09:52:08.334 DEBUG Getting config in section client_rest_transport
09:52:08.334 DEBUG Calling server DELETE http://localhost:59227/api/v1/environment/0e3fb94f-3d20-48bd-b31f-1541547f645a
09:52:08.336 DEBUG HTTP version of request: HTTP/1.1
09:52:08.336 DEBUG Calling method <bound method EnvironmentService.delete_environment of <inmanta.server.services.environmentservice.EnvironmentService object at 0x7f30247d1be0>>(environment_id='0e3fb94...')
09:52:08.339 DEBUG Stopping all autostarted agents for env 0e3fb94f-3d20-48bd-b31f-1541547f645a
09:52:08.501 DEBUG Expiring all sessions for 0e3fb94f-3d20-48bd-b31f-1541547f645a
09:52:08.501 DEBUG Expired session with id 87149f6e-0b51-11ee-a1c8-525400bb0bf2, last seen 0 seconds ago
09:52:08.502 DEBUG Heartbeat wait expired for jenkins-slave-4.ii.inmanta.com, returning. (long poll)
09:52:08.502 INFO 200 POST /api/v1/heartbeat (127.0.0.1) 206.53ms
09:52:08.502 DEBUG expiring session 87149f6e-0b51-11ee-a1c8-525400bb0bf2
09:52:09.343 ERROR An exception occured during the request.
Traceback (most recent call last):
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib/python3.9/site-packages/inmanta/protocol/rest/__init__.py", line 509, in _execute_call
    result = await config.handler(**arguments.call_args)
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib/python3.9/site-packages/inmanta/server/services/environmentservice.py", line 252, in delete_environment
    await self.environment_delete(environment_id)
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib/python3.9/site-packages/inmanta/server/services/environmentservice.py", line 480, in environment_delete
    await asyncio.gather(self.autostarted_agent_manager.stop_agents(env), env.delete_cascade())
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib/python3.9/site-packages/inmanta/data/__init__.py", line 2806, in delete_cascade
    await self.delete()
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib/python3.9/site-packages/inmanta/data/__init__.py", line 2186, in delete
    await self._execute_query(query, *values, connection=connection)
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib/python3.9/site-packages/inmanta/data/__init__.py", line 1611, in _execute_query
    return await con.execute(query, *values)
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib64/python3.9/site-packages/asyncpg/connection.py", line 319, in execute
    _, status, _ = await self._execute(
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib64/python3.9/site-packages/asyncpg/connection.py", line 1658, in _execute
    result, _ = await self.__execute(
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib64/python3.9/site-packages/asyncpg/connection.py", line 1683, in __execute
    return await self._do_execute(
  File "/home/jenkins/workspace/core_inmanta-core_master/project/inmanta-core/.tox/py39/lib64/python3.9/site-packages/asyncpg/connection.py", line 1730, in _do_execute
    result = await executor(stmt, None)
  File "asyncpg/protocol/protocol.pyx", line 201, in bind_execute
asyncpg.exceptions.DeadlockDetectedError: deadlock detected
DETAIL:  Process 2355720 waits for ShareLock on transaction 7975; blocked by process 2355722.
Process 2355722 waits for ShareLock on transaction 7977; blocked by process 2355720.
HINT:  See server log for query details.
09:52:09.343 WARNING Received an exception with status code 500 and message {'message': "An unexpected error occurred in the server while processing the request: ('deadlock detected',)"}
09:52:09.344 ERROR 500 DELETE /api/v1/environment/0e3fb94f-3d20-48bd-b31f-1541547f645a (127.0.0.1) 1008.11ms
``` 

observed here  https://jenkins.inmanta.com/job/core/job/inmanta-core/job/master/1857/

**Comment by arnaudsjs** [comment]
Is this a duplicate of #6090?
